### DISCUSSION TITLE
여러개의 Input 값 다루기

### DISCUSSION THREAD

**Discussion by jaeyeong815** [WantedPreonboardingFE5team/pre-onboarding-7th-2-1-5#31] [Reference]
지난번에 디스코드에서 input의 onChange를 감지할 때 e.target.name과 e.target.value를 사용한다고 말씀드리면서
name을 사용하는 이유는 모르겠고 검색하면 다들 name을 사용해서 따라쓰고 있다고 한 적이 있습니다!

그런데 다른 내용을 찾던 중 React 공식문서에서 권장하는 방법임을 알게되었네요.
[Form Input 공식 문서](https://reactjs.org/docs/forms.html#handling-multiple-inputs)
역시 답은 모두 공식문서에 있나봅니다 =)..

**Comment by baechanghyeon** [comment]
오 이부분에 대해서 궁금했었는데 공식문서에 답이 있었네요! ㅎㅎㅎ 잘 참고 하겠습니다 ! 감사합니다 (_ _)(0 0) !

**Comment by baechanghyeon** [comment]
재영님이 달아 놓으신 공식문서 내용 잘봤습니다 ! 😃
그럼 name을 활용하는 경우로는 다중 Input을 구현 시, 각 Input을 식별하고 그에따라 핸들러 작업을 실행하기 위해 활용한다고 보면 될까요?
다른 블로그([블로그 URL](https://jinchuu1391.tistory.com/50))에서도 보니까 결과적으로 name은 여러개의 Input 식별을 위해 활용한다 라고 결론을 내렸더라구요.. 혹시 이렇게만 이해하면 되는걸까요?... 제가 이해를 잘 한건지 아리송 하네요 🤯

**Reply by jaeyeong815** [reply]
넵 맞습니다 ~~
여러개의 `input`을 각각 `useState()`로 관리한다면 `name`은 필요 없겠지만,
보통 한 `useState()`에 Object로 key, value를 사용해서 한번에 관리하기 때문에, `값이 변한 input`을 식별하기 위해 `name`을 사용하여 `key값을 구별`한다고 보시면 됩니다!


한 `useState()`에 Object로 key, value를 사용해서 한번에 관리하는 형태는
```
const [userData, setUserData] = useState({
  nickname: '',
  age: '',
  adress: '',
})
```
위 형태를 말하는 것 입니다!
여기서 `input`을 건들였을 때 `nickname input`값이 바뀐건지, `age input`값이 바뀐건지 알아내기 위해 `name`을 사용합니다!

**Reply by baechanghyeon** [reply]
오?? 적어주신 예시보니까! 이제 이해가 좀 된거 같습니다 !! 감사합니다~ ㅎㅎ
